### PR TITLE
tree: Remove dead code

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Delete a bunch of dead code, mostly flex tree schema related.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts [hunk 1]
```diff
@@ -151,39 +149,40 @@ export function viewFromState(
...
+	nodeSchemas: Iterable<TreeNodeSchemaIdentifier>,
 	prefix: string,
 	omitInitialNodeSchemas: string[],
-): FlexTreeNodeSchema[] {
-	const values: FlexTreeNodeSchema[] = [];
+): TreeNodeSchemaIdentifier[] {
+	const values: TreeNodeSchemaIdentifier[] = [];
 
-	nodeSchemas.forEach((value, key) => {
+	for (const key of nodeSchemas) {
```
Comment (Josmithr) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts:159]: Nit: should we just call this `schema` or `nodeSchema`? The below `values.push(key)` call is strange :)

**Review by Josmithr — APPROVED** [review]
🥳

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts:159]: its not a schema any more, its just the identifier. This function no longer touches any schema.

File: packages/dds/tree/src/feature-libraries/editableTreeBinder.ts [hunk 2]
```diff
@@ -3,143 +3,12 @@
...
-	B extends OperationBinderEvents | InvalidationBinderEvents,
-> extends DataBinder<B>,
-		Flushable<FlushableDataBinder<B>> {}
+import { type FieldKey, type UpPath, topDownPath } from "../core/index.js";
 
 /**
  * A step in a bind path
  */
-export interface PathStep {
+interface PathStep {
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/feature-libraries/editableTreeBinder.ts:11]: DownPath, which is exported, references PathStep

Comment (CraigMacomber) [reply packages/dds/tree/src/feature-libraries/editableTreeBinder.ts:11]: I considered doing that. We don't require types referenced as implementation details to be exported except at package boundaries. Given this is only used in one place which is a test and we have no plans to use it more, I don't think we really need to export is here (or in the index file)

Comment (CraigMacomber) [reply packages/dds/tree/src/feature-libraries/editableTreeBinder.ts:11]: When trying to trim down dead code, exporting more things that we need makes unused things (like this entire file other than this) hard to find, so I'm erroring on the side of exporting less. Its easier to adding missing exports than remove unneeded ones, so they tend to pile up.

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts (hunk 1 above)
Comment (Josmithr) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts:159]: "identifier" then?

**Pull request merged**